pkg/event/target/elasticsearch.go

```diff
@@ -77,6 +77,20 @@ func (target *ElasticsearchTarget) ID() event.TargetID {
 	return target.id
 }
 
+// IsActive - Return true if target is up and active
+func (target *ElasticsearchTarget) IsActive() (bool, error) {
+	if dErr := target.args.URL.DialHTTP(); dErr != nil {
+		if urlErr, ok := dErr.(*url.Error); ok {
+			// To treat "connection refused" errors as errNotConnected.
+			if IsConnRefusedErr(urlErr.Err) {
```

Comment:
This change is deprecated, we need to use `IsNetworkOrHostDown`. Check the latest changes

Reply:
Since this function `IsActive` has some repeated code, We could replace such places with this function call

Reply:
IsActive has different implementation for different endpoints. Not much of a code is repeated.